data/scripts/sol-flow-node-type-stub-gen.py

```diff
@@ -292,15 +326,53 @@ def declare_packet(outfile, port, packets):
     "NAME": packet_name.upper(),
     "name_data": packet_name + "_packet_data"
     })
+    elif is_composed(data_type):
+        data_type_with_underscore = get_composed_types_with_underscore(data_type)
+        data_type_as_list = get_composed_types_as_list(data_type)
+        outfile.write("""
+static int
+send_%s_packet(struct sol_flow_node *src, uint16_t src_port, %s)
+{
+   struct sol_flow_packet **packets, *packet;
+   const struct sol_flow_packet_type *p_type;
+   uint16_t len, i;
+   int r;
+
+   p_type = %s_get_composed_%s_packet_type();
+   r = sol_flow_packet_get_composed_members_len(p_type, &len);
+   SOL_INT_CHECK(r, < 0, r);
+   packets = calloc(len, sizeof(struct sol_flow_packet *));
+   SOL_NULL_CHECK(packets, -ENOMEM);
+
+   %s
+
+   packet = sol_flow_packet_new(p_type, packets);
```

Comment:
your code does `packets = calloc() -> sol_flow_packet_new(p_type, packets)`, so packets is actually all zeros. If you give this it will propagate to internal `sol_flow_packet_new(sub_type, sub_packet_init)`, with `sub_packet_init` being all zero. This is likely wrong.

What we should do is handle `packets` and `sub_packet_init` being NULL pointers, then you don't need to call `packets = calloc(...)`, only call `packet = sol_flow_packet_new(p_type, NULL)`.